mne/time_frequency/tfr.py

```diff
@@ -245,6 +245,248 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
+        raise ValueError('sfreq must be a float or an int.')
+    sfreq = float(sfreq)
+
+    # Check output
+    allowed_ouput = ('complex', 'power', 'phase',
+                     'avg_power_itc', 'avg_power', 'itc')
+    if output not in allowed_ouput:
+        raise ValueError("Unknown output type. Allowed are %s but "
+                         "got %s" % (allowed_ouput, output))
+
+    if method not in ('multitaper', 'morlet'):
+        raise ValueError('method must be "morlet" or "multitaper"')
+
+    # Default zero_mean = True if multitaper else False
+    zero_mean = method == 'multitaper' if zero_mean is None else zero_mean
+    if not isinstance(zero_mean, bool):
+        raise ValueError('')
+    frequencies = np.asarray(frequencies)
+
+    # XXX Can we compute single-trial phases with multitaper?
+    if (method == 'multitaper') and (output == 'phase'):
+        raise NotImplementedError(
+            'This function is not optimized to compute the phase using the '
+            'multitaper method. Use np.angle of the complex output instead.')
+
+    # Setup wavelet
+    if method == 'morlet':
+        W = morlet(sfreq, frequencies, n_cycles=n_cycles, zero_mean=zero_mean)
+        Ws = [W]  # to have same dimensionality as the 'multitaper' case
+        if time_bandwidth != 4.0:
+            raise ValueError('time_bandwidth only applies to "multitaper"'
+                             ' method.')
+    elif method == 'multitaper':
+        Ws = _make_dpss(sfreq, frequencies, n_cycles=n_cycles,
+                        time_bandwidth=time_bandwidth, zero_mean=zero_mean)
+
+    # Check wavelets
+    if len(Ws[0][0]) > epoch_data.shape[2]:
+        raise ValueError('At least one of the wavelets is longer than the '
+                         'signal. Use a longer signal or shorter wavelets.')
+
+    # Initialize output
+    decim = _check_decim(decim)
+    n_freqs = len(frequencies)
+    n_epochs, n_chans, n_times = epoch_data[:, :, decim].shape
+    if output in ('power', 'phase', 'avg_power', 'itc'):
+        dtype = np.float
+    elif output in ('complex', 'avg_power_itc'):
+        # avg_power_itc is stored as power + 1i * itc to keep a
+        # simple dimensionality
+        dtype = np.complex
+
+    if ('avg_' in output) or ('itc' in output):
+        out = np.empty((n_chans, n_freqs, n_times), dtype)
+    else:
+        out = np.empty((n_chans, n_epochs, n_freqs, n_times), dtype)
+
+    # Parallel computation
+    parallel, my_cwt, _ = parallel_func(_time_frequency_loop, n_jobs)
+
+    # Parallelization is applied across channels.
+    tfrs = parallel(
+        my_cwt(channel, Ws, output, use_fft, 'same', decim)
+        for channel in epoch_data.transpose(1, 0, 2))
+
+    # FIXME: to avoid overheads we should use np.array_split()
+    for channel_idx, tfr in enumerate(tfrs):
+        out[channel_idx] = tfr
+
+    if ('avg_' not in output) and ('itc' not in output):
+        out = out.transpose(1, 0, 2, 3)
```

Comment:
This is to enforce that the output is always `epochs x chans x ...` right? Might be worth a quick comment to that effect since there's a lot of transposing going on here to save future confusion.